nix/lint.nix

```diff
@@ -0,0 +1,12 @@
+{ writeShellScriptBin, git, silver-searcher, hlint }:
+
+writeShellScriptBin "postgrest-lint"
+  ''
+    set -euo pipefail
+
+    rootdir="$(${git}/bin/git rev-parse --show-toplevel)"
+
+    # Lint Haskell files
+    ${silver-searcher}/bin/ag -l -g '\.l?hs$' "$rootdir" \
+      | xargs ${hlint}/bin/hlint -X QuasiQuotes -X NoPatternSynonyms
```

Comment:
Here  you use the `silver-searcher` instead of `git ls-files | grep '\.l\?hs$$'` because the command is shorter right?

I was thinking just using `git` and `grep` would avoid having an extra dependency, but I think is fine since we use Nix here.

Reply:
There is a small difference, this will also pick up files that are not yet staged in git, while still respecting .gitignore etc. That might be useful for contributors if they are working on adding new files.

Reply:
That's better! I've already tested it and it's working good. Merging..